doc/sphinx-guides/source/developers/dev-environment.rst

```diff
@@ -294,6 +294,72 @@ If you have an old copy of the database and old Solr data and want to start fres
 
 You may also find https://github.com/IQSS/dataverse/blob/develop/scripts/deploy/phoenix.dataverse.org/deploy and related scripts interesting because they demonstrate how we have at least partially automated the process of tearing down a Dataverse installation and having it rise again, hence the name "phoenix." See also "Fresh Reinstall" in the :doc:`/installation/installation-main` section of the Installation Guide.
 
+Provenance
+----------
+
+The provenance feature of Dataverse depends on a REST service which you must install and configure. This is similar to how Solr provides an REST API that Dataverse is configured to use. The provenance REST service is built on top of "prov-cpl" (Core Provenance Library or CPL) and the code can be found at https://github.com/ProvTools/prov-cpl .
+
+Installing CPL on Mac
+~~~~~~~~~~~~~~~~~~~~~
+
+If you're feeling adventurous, you can attempt to install CPL directly on your Mac but this is not recommended. Rather, below CPL runs as a REST service within Vagrant.
+
+First, install Vagrant and VirtualBox as described in the :doc:`tools` section.
+
+Then, clone the ``prov-cpl`` repo.
+
+git clone https://github.com/ProvTools/prov-cpl
+
+Download :download:`Vagrantfile <../_static/developers/prov/Vagrantfile>` and :download:`vagrant.sh <../_static/developers/prov/install/vagrant.sh>` and place them in the ``prov-cpl`` directory that was created when you cloned that repo. Then ``cd`` to that directory and run ``vagrant up``.
+
+``vagrant up`` is expected to take a while. A message near the end should say ``Running on http://0.0.0.0:5000/`` which indicates that the CPL REST service is running within Vagrant. The Vagrantfile above specifies that port 5000 within Vagrant should be available to your laptop on port 7777. To test this, try the following curl command from your laptop:
+
+``curl http://localhost:7777/provapi/version``
+
+If you get an HTTP response and JSON output, you have successfully set up the REST service. The next step is to configure Dataverse to use it.
+
+curl -X PUT -d 'http://localhost:7777' http://localhost:8080/api/admin/settings/:ProvServiceUrl
+
+Installing CPL on Ubuntu
```

Comment:
As someone who is relatively familiar with Vagrant, I would disagree with documenting the version of Ubuntu because what matters is the "base box" in the Vagrantfile. This is currently `config.vm.box = "ubuntu/xenial64"` which means 16.04 (the latest LTS as of this writing). Basically a comment here has the potential to get out of date. That said, I'm fine adding a comment if it's truly helpful.